### ISSUE TITLE
Binding going offline when linking Channel Homemode Switch

### ISSUE THREAD

**Issue by TsAbsinthe** [nibi79/synologysurveillancestation#56]
Hello, 

the Binding is going offline when linking Channel Homemode Switch.

<img width="265" alt="Bildschirmfoto 2023-10-11 um 21 35 54" src="https://github.com/nibi79/synologysurveillancestation/assets/112728032/6f934357-c208-45d9-8cb6-f1e8140a1a0b">

Any idea how to solve this problem? 

openhab 4.0.2

**Comment by Pavion** [comment]
Hi and sorry for I can't reproduce your issue. 
With the marketplace version (3.0.0.xxx) I don't have any problems. With the latest GitHub release 4.0.0.202304020711 and OH4.1 I can't create new Things anymore (why?) but the Things created with 3.0.0 or a text file are working fine. Could you please provide your binding's version (using Karaf's `bundle:list`)? You can also enable debug mode using:
```
log:set DEBUG org.openhab.binding.synologysurveillancestation
```
.. so you might see an additional info / exception (don't forget to remove it afterwards with `log:set ERROR ...`  to avoid log cluttering).
I'm not using this binding much for myself anymore but if you provide some info I'll take a look around this weekend.

**Comment by TsAbsinthe** [comment]
Hi, first - sorry for the late response. 

In the log are several entries... 

`2023-10-24 21:12:44.049 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler SynoCameraHandler of thing synologysurveillancestation:camera:b4596bb2b1:7be07e0b17 tried checking if channel common#enable is linked although the handler was already disposed.
2023-10-24 21:12:44.050 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler SynoCameraHandler of thing synologysurveillancestation:camera:b4596bb2b1:7be07e0b17 tried checking if channel common#record is linked although the handler was already disposed.
2023-10-24 21:12:44.050 [WARN ] [.core.thing.binding.BaseThingHandler] - Handler SynoCameraHandler of thing synologysurveillancestation:camera:b4596bb2b1:7be07e0b17 tried checking if channel common#snapshot-uri-dynamic is linked although the handler was already disposed.`

The same are repeating all the time alle few seconds. 

Any idea? Or do you need more or other informations? 

Greetings

**Comment by Pavion** [comment]
Welcome back! 

It looks like there is something broken on some update. 
Could you please check:

1. Whether you have but one version of this binding running with `bundle:list | grep Syno` in Karaf. You should see but one entry with version like `4.0.0.202304020711` 

if it's valid, 

2. Please try to recreate your Things and Items (perhaps on a fresh downloaded test setup first) and see, whether the error persists. 

BR Pav

**Comment by TsAbsinthe** [comment]
Hi Pav, 

thanks for the fast response. It looks valid: 

301 │ Active │  80 │ 4.0.0.202304021037     │ openHAB Add-ons :: Bundles :: Synology Surveillance Station Binding    

I delete the bridge and all items. Reboot the openhab an created the bridge and the HomeMode-Item new again... same error... 

Any other ideas?

**Comment by Pavion** [comment]
Sorry, I have just checked again and still can not reproduce it. 
I was using a clean openhab-4.1.0-SNAPSHOT.zip installation with just the latest jar in the `/addons` folder.
Created a Thing with proper credentials, linked the HomeMode channel and was able to use it with no errors.
Have you too tried a clean installation? You could try to create a dedicated Surveillance Station user to check whether it's a rights issue. 
If it's still not working, please enable debug mode (see my second reply) and attach a full log from Thing creation to Item switching failing (please note that your password will be logged as plain text, consider to replace it with dummy text first).

**Comment by TsAbsinthe** [comment]
Hi Pav, 

I found it. As always... Error 40 ;) 

If you enter the IP in the configuration with the Portnumber it will crash ;) 

Ex.: 192.168.178.40:3000

**Comment by Pavion** [comment]
I'm almost afraid to ask what you had entered into "port" field 🤣  
Glad you've solved it and thanks for reporting back

**Comment by TsAbsinthe** [comment]
I entered the right port. But then the binding isn't "intelligent" enough to correct the user ;) 

so the binding ist trying to connect to the syno with nonsense like https://[http://192.170.160.1:3000/]:3000

But, the error first occur when I linked a channel. Before the status is online. But it can't work out - so this was the misleading info... I thought the binding is working correct and the item is the problem...
